src/api/ipfs.utils.ts

```diff
@@ -0,0 +1,13 @@
+import { throwError } from 'util/errors'
+
+const isIpfsCid = (addr: string) => /^(Qm[a-zA-Z0-9]{44})$/.test(addr)
+
+export const isIpfsProtocol = (addr: string) => /^ipfs:\/\/(.*)$/.test(addr)
```

Comment:
Also wondering if we should be able to identify that it is an ipfs hash and allow:

```
QmRtUNf7weCtsHKCbkTQyrhZHAmsrMNPdmSPZjWFRJVRsW
```

```
Unexpected token 'Q', "QmRtUNf7we"... is not valid JSON
```

Reply:
Also related, I removed the metadata by replacing it with an empty string and received:

```
Unexpected end of JSON input
```

Reply:
I think it's appropriate to assume there will always be either JSON, or a URI w/ a scheme (e.g. "ipfs://", "http://", or "https://").  imo a raw IPFS hash without a scheme (e.g. `QmRtUNf7weCtsHKCbkTQyrhZHAmsrMNPdmSPZjWFRJVRsW`) is not something we should support.


The reading of off chain metadata ipfs IPFS should just involving supporting the `ipfs://` scheme for a URI, by querying via the `ipfsClient` that is setup in this PR.

I'm not opposed to us also adding support for `http://` and `https://` schemes, but would consider it a nice-to-have, and not essential. It's not documented anywhere in the [x/group docs](https://github.com/cosmos/cosmos-sdk/tree/main/x/group) as a valid type of metadata, so it might be cleaner to not support it at all. But then again, EVM ecosystems typically do support HTTP URIs in addition to IPFS URIs so so maybe its appropriate for us to do the same.

If we do want to support HTTP/HTTPS, the logic should follow the same logic as IPFS schemes: 
- retreive off-chain data in accordance w/ the scheme (e.g. querying `ipfsClient`, or making an HTTP GET request)
- parsing the response body JSON into structured metadata